### ISSUE TITLE
Can not import Quill(.min).js into Salesforce's lightning component

### ISSUE THREAD

**Issue by splacentino** [quilljs/quill#1250]
Hello,

Since two weeks I am using Quill into one of my lightning component (Salesforce) without any issues by importing it as this:

Component.cmp

```html
  <ltng:require
    scripts="{! join(',',
          $Resource.lodash, $Resource.quill
    )}"
    styles="{!
            $Resource.quillcss
    }"
    afterScriptsLoaded="{! c.scriptLoaded }"
  />
```

controller.js:

```js
scriptLoaded: function(component) {
  const editor = component.find('editor');

  //const element = _.get(editor, 'elements[0]'); // this does not work anymore
  const element = editor.getElement();

  if (element) {
    component.set('v.quill',
      new Quill(element, {
        placeholder: 'Mail content..',
        theme: 'snow',
        modules: {
           toolbar: [
                [{ 'header': [1, 2, 3, 4, 5, 6, false] }],
                ['blockquote', 'code-block'],
                ['bold', 'italic', 'underline', 'strike'],
                [{ 'list': 'ordered'}, { 'list': 'bullet' }],
                ['link'],
                ['clean'],
                [{'color' : [ 'blue','red','green','white','yellow','black','purple','pink']}]
          ]
        }
      })
      );
      component.set('v.scriptLoaded', true);
  } else {
    console.error('No container for text area.');
  }
}
```

And everything worked well until last Saturday's update (07th of January) and its `lockerservice` activation. Now, the Quill script can not be loaded correctly and so I am not able to use it anymore.

I thing the issue is this line
```js
if (window.katex == null) {
  throw new Error('Formula module requires KaTeX.');
}
```

from http://cdn.quilljs.com/1.1.9/quill.js

(Cf. [Salesforce documentation about Secure window](https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/cli_rules_secure_window.htm)

Are you publishing soon a Salesforce compliant version of Quill with a secure usage of `window` element ?

**Version**:

`Quill version`: 1.1.9

**Comment by jhchen** [comment]
For Salesforce platform specific, please contact Salesforce. The line referenced (`window.katex`) is only run when you use the [formula](http://quilljs.com/docs/modules/formula/) module, which you are not. Is the from a stack trace (if so please post that)?

**Comment by splacentino** [comment]
@jhchen

Hello, no stacktrace available. Ill send a message to a salesforce lightning dev today but Im hundred % sure that the script is "not well written for salesforce" as lot are..

Thank you anyway

**Comment by splacentino** [comment]
It supposed to be a Salesforce issue (Cf. [Stackexchange discussion about w/ Doug Chasman](http://salesforce.stackexchange.com/questions/155236/lightning-component-scripts-loading-issues-since-spring-17-update/155264#155264)).

**Comment by mitfity** [comment]
@splacentino It is strange. I get scripts loaded, but when I try to initialize Quill — I get this:
[Parchment] Unable to create SecureElement: [object HTMLDivElement]{ key: {"namespace":"c"} } blot
However, everything works without LockerService.

**Comment by splacentino** [comment]
@mitfity Here is an piece of code I am using:

```javascript
      const editor    = component.find('editor'); // where 'editor' is a <div aura:id="editor" />
      const element = editor.getElement();
      const quill = new Quill(
        element,
        {
          placeholder: 'Mail content..',
          theme: 'snow',
          modules: {
            toolbar: [
              [{ 'header': [1, 2, 3, 4, 5, 6, false] }],
              ['blockquote', 'code-block'],
              ['bold', 'italic', 'underline', 'strike'],
              [{ 'list': 'ordered'}, { 'list': 'bullet' }],
              ['clean'],
              [{'color' : [ 'blue', 'red', 'green', 'yellow', 'black', 'purple', 'pink']}]
            ]
          }
        }
      );
      quill.on('text-change', function() {
                // Do stuff with quill.container.firstChild.innerHTML;
      });
      component.set('v.quill', quill); // where quill is an attribute of type Object
```

Do not forget to do this *ONLY* on the callback when the script is loaded:
```html
  <ltng:require scripts="{! $Resource.quill }" afterScriptsLoaded="{! c.scriptLoaded }" />
```

Hope this will help you.

**Comment by mitfity** [comment]
@splacentino What is $Resource.quill you import? I do the same but get error.

**Comment by mikydiamantstein** [comment]
Did you have any luck with it (with LockerSevice enabled)?

**Comment by mitfity** [comment]
@mikyd I actually got answer from salesforce development team and they said "To my knowledge we do not support quillJs directly." and that they are working on native richtext editor.
"For transparency, we believe our past tool in classic was used for not only editing RTE fields but also as a broader authoring tool. We believe these are separate use cases and require separate solutions. As for RTE editing, we are working on a comprehensive, long term approach for a WYSIWG editor that is both secure (locker compliant) and meets our customers needs."

**Comment by splacentino** [comment]
@mitfity Yet this is the library I uploaded on SF as static resource.
It is compliant w/ Lockerservice (you can browse the Quill code and compare it to Lockeservice restrictions (what I did)).

It HAS to be supported because the `<Lightning:inputRichText />` is based on QuillJS.
https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/aura_compref_lightning_inputRichText.htm
> A lightning:inputRichText component creates a rich text editor based on the Quill JS library,

**Comment by mitfity** [comment]
@splacentino I don't know where you use Quill and did you do anything with quill.js so it will work with LockerService. I tried again in different org and I am getting:
`[[Parchment] Unable to create SecureElement: [object HTMLDivElement]{ key: {"namespace":"extdev"} } blot]` , where extdev is my org namespace. This error goes from quill.js and as I see this is because LockerService adds namespace now, so quill doesn't understand what html element it is.
I tried plenty of time and maybe because I am new to Salesforce I don't get it and I don't know how to solve this.
